### DISCUSSION TITLE
Listing States missing from API spec

### DISCUSSION THREAD

**Discussion by Gareth064** [etsy/open-api#1177] [Q&A] [answered]
I'm in the process of syncing Etsy receipts , going back to 2016, to my local DB and I want to get the the listings, associated to the receipts, using the Listing ID's.

I have a JSON converter which converts the string value to an 'Listing State' Enum.

![image](https://github.com/etsy/open-api/assets/22307223/e57ae762-33fa-48e1-a295-57e900a1cb70)

My Enum is built using the document Listing State values.

`active` `inactive` `sold_out` `draft` `expired`

![image](https://github.com/etsy/open-api/assets/22307223/f47ad8d1-59d3-4738-97c6-796adb15227c)

Some of my receipts are related to Listings which are returning some Listing States which are not documented. 

`edit` and `removed`

This was a simple fix for me as I just updated my Enum to include these, but for the benefit of the community, it would be good if these states were documented along with what they mean. I get what `removed` likely means as these are really old orders and we might have deleted those listings since, but not sure what `edit` would mean.

This discussion is simply here to let the community know about these undocumented states. If you are one of the many developers who use Code Generators to create you Etsy API Clients using the Spec, then you will have the task of adding these States in manually every time you regenerate the code.

**Comment by simonj** [comment]
I also was getting this "edit" but it seems like the Edit is equal to draft

**Comment by showtroylove** [comment]
I identified another enum problem with Etsy's dynamic enum values. If you've observed any issues with new year's activity and have utilized an enum for the `when_made` field, there might be a problem. I reverted the type back to a string because the field is not useful internally. They could have specified a date field, and displayed it the desired string format.  Moral of the story... don't use enum with Etsy API.

**Comment by yovasx2** [comment]
i have also seen the new states, those are causing problem for me as well

**Comment by franciscodelgadodev** [comment]
I've noticed it also affects API state filters and responses.

For example, if I make an API call to get some `active` listings (`GET https://api.etsy.com/v3/application/shops/<shopId>/listings?limit=100&state=active`) those items with `edit` status are returned in the response.

The same happens regardless of the state I send in the API request. Listings in `edit` state are returned when I send API calls with `status=active`, `status=draft` or `status=inactive`.

Some documentation around this would also be very helpful.

Thanks!

**Comment by joaquin102** [comment]
This is still a problem. I have a listing that I literally deleted and I receive the state as `edit`

Does anyone know how to identify a deleted listing?

**Comment by franciscodelgadodev** [comment]
Any word on this, Etsy team? It's still happening to many users on our end...

**Comment by openapi-support** [comment]
We are actively investigating issues with ListingState.

**Comment by will-blanco** [comment]
Hi Etsy team,

Any word on this, Etsy team? It's still happening to many users on our end...

[a comment by an automated account is not shown]
